### ISSUE TITLE
Ignore backups does not include core

### ISSUE THREAD

**Issue by tjorim** [sabeechen/hassio-google-drive-backup#600]
Hi there,

By default Home Assistant creates backups of core and addons when upgrading.
Handy in some cases but cumbersome to clean it up once in a while.

I enabled the following two settings in this addon so it does ignore the "upgrade" backups and deletes them after 3 days.

![Screenshot_20220430-044437.jpg](https://user-images.githubusercontent.com/1294876/166087607-6df764b3-fb61-4f2c-8c95-8c603e59b4b2.jpg)

This works fine for most addons, but does not work for Home Assistant 'core' backups.

![Screenshot_20220430-044453.jpg](https://user-images.githubusercontent.com/1294876/166087681-a9cbe0d9-6f50-4108-b63f-efd3e9de21ee.jpg)

**Comment by sabeechen** [comment]
Do the "core" backups appear in the addon's web-ui at the bottom under "Ignored backups"?  I've been using this option personally for a while now and haven't seen ti fail to recognize like this.

**Comment by SSMCD** [comment]
I'm having the same issue.  The HA core-only backups are being counted towards the generational backups.  They do not show up under "Ignored backups" in the UI.

At first I thought this line might still be returning true:
`single_backup = len(self.details().get("addons", [])) + len(self.details().get("folders", [])) == 1`

But when the core-only backup is clicked in the UI, nothing shows up where the backup "Contents" should be - no folders or addons are listed.

**Comment by encryptix** [comment]
I am also seeing the same for core backups and had to turn on "ignore other backups"

Ignoring "Upgrade" backups wasnt doing it. With ignore other backups on core_  backups are appearing in the ignored section

**Comment by sabeechen** [comment]
The Home Assistant config folder doesn't show up in the "contents" view of the addon, which I think might be a bug.  It could be that nothing is in the snapshot (eg folders+addons == 0), though then it should show up as _very_ small, like under 1kb.  Is that the case?  I'm not even sure Home Assistant can create a backup like that.

Could I convince one of you to share one of these problematic "core" backups with me?  that would save _a lot_ of time going back and forth trying to figure out what makes these snapshots special.  I understand that might mean sharing sensitive info with me, but I'll treat it with care (eg overwite it with zeros once I complete my testing, wipe the VM used for testing, etc).  If so I'll email you a drop folder on Google Drive to upload into.

If not could I have you extract the contents of one of these backups (its just a standard tar file) and post the contents of backup.json file inside here?  This is the backup metadata file and shouldn't have any sensitive info in it.

**Comment by SSMCD** [comment]
The core backup contains my entire /config directory (running hassos) which is about 70 MB of data (~16.3 MB compressed).
I created the backup from Home Assistant by selecting "Partial backup" and only checking off "Home Assistant
(2022.5.2)"


![Screenshot 2022-05-07 140129](https://user-images.githubusercontent.com/21278439/167266366-fbe95261-208b-4875-b7d5-cee5b123ce58.png)

The backup folder structure looks like this:
```
Backup (named May 7, 2022)
-> backup.json
-> homeassistant.tar.qz
---> homeassistant.json
---> data (this folder contains the entire /config directory)
```

backup.json contains the following:

```
{
  "version": 2,
  "slug": "88f76e53",
  "name": "May 7, 2022",
  "date": "2022-05-07T12:45:20.312119+00:00",
  "type": "partial",
  "addons": [],
  "protected": false,
  "compressed": true,
  "repositories": [
    "https://github.com/hassio-addons/repository",
    "https://github.com/sabeechen/hassio-google-drive-backup"
  ],
  "homeassistant": {
    "version": "2022.5.2",
    "size": 16.3
  },
  "docker": {
    "registries": {}
  },
  "folders": [],
  "crypto": null
}
```

**Comment by sabeechen** [comment]
Thank you, that explains it, I can make a fix for this.

**Comment by sabeechen** [comment]
Oof, looks like this has always been busted.  Why didn't anyone tell me until now!?  I'll have a fix out soon.

**Comment by sabeechen** [comment]
I just released v0.107.2 which includes the fix to correctly recognize core-only backups as upgrade backups.  I also fixed the details page in the addon web-ui so it correctly indicates if the Home Assistant config folder is present in a backup.  If you run into problems after updating, please feel free to reopen this issue or let me know.
